openmmtools/multistate/sams.py

```diff
@@ -234,6 +250,17 @@ def __init__(self,
         self._replica_neighbors = None
         self._cached_state_histogram = None
 
+    @property
+    def _criteria(self):
+        return { criteria: value for criteria, value in (('histogram_flatness', self.histogram_flatness),
+                                                         ('minimum_round_trips', self.minimum_round_trips),
+                                                         ('minimum_logZ', self.minimum_logZ),
+                                                         ('minimum_visits', self.minimum_visits))
+                                                         if value is not None}
+    @property
+    def _advance(self):
+        return {criteria: False for criteria, value in self._criteria.items() if value is not None}
```

Comment:
I'm not sure this is going to work. Here you're regenerating an all-false dictionary every time you call `self._advance` so doing `self._advance[criteria] = True` won't actually change it.

Actually, if I understand the code correctly, you can even remove this as an object variable/property and simply initialize an `advance` local variable inside `_update_stage()` before checking all criteria.

Reply:
After merging master, the version that it is being installed is 0.16, is this correct?

Reply:
Hmm. Are you having those problems locally? On Travis, it looks like versioneer is installing 0.18.2 as expected: https://travis-ci.org/choderalab/openmmtools/jobs/542376730#L884

Reply:
Yes, on lilac.

Reply:
I see, if you're installing from source, it shouldn't matter which version is printed as long as the installed code is updated. Unless I'm misunderstanding the problem.